binaries/cuprated/src/signals.rs

```diff
@@ -0,0 +1,9 @@
+//! Signals for Cuprate state used throughout the binary.
+
+use tokio::sync::RwLock;
+
+/// Reorg lock.
+///
+/// A [`RwLock`] where a write lock is taken during a reorg and a read lock can be taken
+/// for any operation which must complete without a reorg happening.
+pub static REORG_LOCK: RwLock<()> = RwLock::const_new(());
```

Comment:
I have mixed feelings using locks in a C-style like this... The solution is to design things such that you can scope and pass resources properly, I'm not going to ask this to be changed since that will involve a lot of work but I hope you realize this pattern is bad as well.

Reply:
related issue: https://github.com/Cuprate/cuprate/issues/305